commons/zenoh-codec/src/transport/init.rs

```diff
@@ -311,6 +326,10 @@ where
             n_exts -= 1;
             self.write(&mut *writer, (qos, n_exts != 0))?;
         }
+        if let Some(qos_optimized) = ext_qos_optimized.as_ref() {
```

Comment:
Same comment as above: both `ext_qos` and `ext_qos_optimized` could be sent on the network.
If `ext_qos_optimized` then I don't see much the need of sending as well `ext_qos`.

Reply:
See above.